snappy_test.go

```diff
@@ -375,3 +390,19 @@ func Benchmark_ZFlat8(b *testing.B)  { benchFile(b, 8, false) }
 func Benchmark_ZFlat9(b *testing.B)  { benchFile(b, 9, false) }
 func Benchmark_ZFlat10(b *testing.B) { benchFile(b, 10, false) }
 func Benchmark_ZFlat11(b *testing.B) { benchFile(b, 11, false) }
+func Benchmark_ZFlat12(b *testing.B) { benchFile(b, 12, false) }
+
+// Prints compression size and ratio.
+func BenchmarkCompressionSize(b *testing.B) {
+	fmt.Println("\ndata\tinsize\toutsize\treduction")
+	for n := range testFiles {
+		if err := downloadTestdata(b, testFiles[n].filename); err != nil {
+			b.Fatalf("failed to download testdata: %s", err)
+		}
+		src := readFile(b, filepath.Join(*testdata, testFiles[n].filename))
+		dst := make([]byte, MaxEncodedLen(len(src)))
+		dst = Encode(dst, src)
+		fmt.Printf("%s\t%d\t%d\t%.2f%%\n", testFiles[n].label, len(src), len(dst), 100-float64(len(dst))/float64(len(src))*100)
+	}
+	b.Skip("ok")
```

Comment:
Perhaps this is worth a little comment.
// BenchmarkCompressionSize isn't really a benchmark, in the sense of "I
// want to run some code b.N times and see how long it takes". Instead, it
// prints out compressed sizes for the set of testFiles. Like the ns/op or
// MB/s metrics that the Benchmark_Z* benchmarks above give, this
// compression metric is also useful to know when tweaking the encoding
// algorithm, so this function is also run as a 'benchmark'.
b.Skip("ok")